jedi/parser/tree.py

```diff
@@ -52,6 +53,19 @@ def is_node(node, *symbol_names):
         return type in symbol_names
 
 
+def _fix_forward_reference(annotation):
+    if isinstance(annotation, String):
```

Comment:
>  since type_factory(list) is probably not allowed in pep-0484

It is (PEP 484, Forward References): 

> The string literal should contain a valid Python expression (i.e., compile(lit, '', 'eval') should be a valid code object) 

-----

Testing it should be straight forward, e.g.: 

```
def test_forward_references(x: wrap_reference(), y: 'wrap_reference()'):
    """
    x cannot be resolved, because it's out of scope.
    y however can be resolved, because it's a string.
    """
    #?
    x
    #? int()
    y
```